### ISSUE TITLE
Segmentation fault when toggling video stream

### ISSUE THREAD

**Issue by Dudemanguy** [mpv-player/mpv#11088]
### Important Information

- mpv version: starting from cb15bc4324327bded956b2c9b78f09bdc2729d98
- Linux Distribution and Version: Arch/Artix Linux

### Reproduction steps

- Play any video: `mpv --no-config input.mkv`
- Toggle the video on/off (e.g. press `_` twice)

### Expected behavior

Video window goes away and then comes back.

### Actual behavior

Segfault when trying to toggle the video on again.
```
video: no
: 00:00:00 / 00:00:31 (1%)
[mkv] Discarding potentially broken or useless index.
zsh: segmentation fault  ./build/mpv --no-config ~/src/sample-video/cceijly.mkv --log-file=out.log
```

### Log file
[Core dump](https://0x0.st/oRJg.txt)
[Not working log](https://0x0.st/oRJ6.log)
[Working log](https://0x0.st/oRJI.log)

**Comment by Dudemanguy** [comment]
Oh I'm stupid. This only happens with mkv and h264 because `init_avctx` has some old workaround apparently.

**Comment by sfan5** [comment]
This?
https://github.com/mpv-player/mpv/blob/3a6ef8170551e8eab9ef89143db04e2e0d7ec567/video/decode/vd_lavc.c#L766-L776

**Comment by Dudemanguy** [comment]
Yeah. The first time you initialize a file, it's fine but toggling it back on cause `avcodec_send_packet` to segfault there. I assume there's something in `mp_set_av_packet` that needs to be changed but I'm not sure what.

**Comment by Dudemanguy** [comment]
Fixed by 33e73f4efde974584c9b577ea31f4441ae4545b5.
